server/src/main/java/io/druid/segment/indexing/DataSchema.java

```diff
@@ -46,6 +50,9 @@ public DataSchema(
       @JsonProperty("granularitySpec") GranularitySpec granularitySpec
   )
   {
+    Preconditions.checkNotNull(dataSource, "dataSource cannot be null. Please provide a dataSource.");
+    Preconditions.checkNotNull(aggregators, "metricsSpec cannot be null. Please provide a metricsSpec.");
```

Comment:
why not allow a segment without metrics? I might just want to count rows with a given filter.